literals.go

```diff
@@ -0,0 +1,777 @@
+func (t TimestampLiteral) Type() Type                      { return PrimitiveTypes.Timestamp }
+func (t TimestampLiteral) Value() Timestamp                { return Timestamp(t) }
+func (t TimestampLiteral) String() string {
+	tm := time.UnixMicro(int64(t)).UTC()
+	return tm.Format("2006-01-02 15:04:05.000000")
+}
+func (t TimestampLiteral) To(typ Type) (Literal, error) {
+	switch typ.(type) {
+	case TimestampType:
+		return t, nil
+	case TimestampTzType:
+		return t, nil
+	case DateType:
+		tm := time.UnixMicro(int64(t)).UTC()
+		return DateLiteral(tm.Truncate(24*time.Hour).Unix() / int64((time.Hour * 24).Seconds())), nil
+	}
+	return nil, fmt.Errorf("%w: TimestampLiteral to %s", ErrBadCast, typ)
+}
+func (t TimestampLiteral) Equals(other Literal) bool {
+	return literalEq(t, other)
+}
+
+type StringLiteral string
+
+func (StringLiteral) Comparator() Comparator[string] { return cmp.Compare[string] }
+func (s StringLiteral) Type() Type                   { return PrimitiveTypes.String }
+func (s StringLiteral) Value() string                { return string(s) }
+func (s StringLiteral) String() string               { return string(s) }
+func (s StringLiteral) To(typ Type) (Literal, error) {
+	switch t := typ.(type) {
+	case StringType:
+		return s, nil
+	case Int32Type:
+		n, err := strconv.ParseInt(string(s), 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("%w: casting '%s' to %s",
+				errors.Join(ErrBadCast, err), s, typ)
+		}
+
+		if math.MaxInt32 < n {
+			return Int32AboveMaxLiteral(), nil
+		} else if math.MinInt32 > n {
+			return Int32BelowMinLiteral(), nil
+		}
+
+		return Int32Literal(n), nil
+	case Int64Type:
+		n, err := strconv.ParseInt(string(s), 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("%w: casting '%s' to %s",
+				errors.Join(ErrBadCast, err), s, typ)
+		}
+
+		return Int64Literal(n), nil
+	case Float32Type:
+		n, err := strconv.ParseFloat(string(s), 32)
+		if err != nil {
+			return nil, fmt.Errorf("%w: casting '%s' to %s",
+				errors.Join(ErrBadCast, err), s, typ)
+		}
+		return Float32Literal(n), nil
+	case Float64Type:
+		n, err := strconv.ParseFloat(string(s), 64)
+		if err != nil {
+			return nil, fmt.Errorf("%w: casting '%s' to %s",
+				errors.Join(ErrBadCast, err), s, typ)
+		}
+		return Float64Literal(n), nil
+	case DateType:
+		tm, err := time.Parse("2006-01-02", string(s))
+		if err != nil {
+			return nil, fmt.Errorf("%w: casting '%s' to %s - %s",
+				ErrBadCast, s, typ, err.Error())
+		}
+		return DateLiteral(tm.Truncate(24*time.Hour).Unix() / int64((time.Hour * 24).Seconds())), nil
+	case TimeType:
+		val, err := arrow.Time64FromString(string(s), arrow.Microsecond)
+		if err != nil {
+			return nil, fmt.Errorf("%w: casting '%s' to %s - %s",
+				ErrBadCast, s, typ, err.Error())
+		}
+
+		return TimeLiteral(val), nil
+	case TimestampType:
+		val, err := arrow.TimestampFromString(string(s), arrow.Microsecond)
```

Comment:
In Iceberg we are quite strict around our timezone handling.  A TimestampType should have no timestamp,  TimestampTzType should have mentioned the timestamp explicitly. See https://github.com/apache/iceberg-python/blob/42afc439d362ef1b3dcff03a1ffd959bc0a399ca/pyiceberg/utils/datetime.py#L29-L33

Reply:
@zeroshade can you fix this one before merging?